### PR TITLE
Update to python-telegram-bot V12

### PR DESCRIPTION
**Pull request by qttylog**

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -56,6 +65,7 @@ def infos(mapping):
         "/allergens": "Allergene auswählen.",
         "/resetallergens": "Allergene zurücksetzen",
         "/info": "Informationen über gewählte Mensa, Abonnement und Allergene.",
+        "/status": "Status des Bots",
```
Comment (kmein) [inline bot.py]: I wouldn't show this to the users.

File: bot.py [hunk 2]
```diff
@@ -41,7 +47,10 @@
 user_db = MenstruationConfig(REDIS_HOST)
 
 
-def help_handler(bot: Bot, update: Update):
+def help_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: help_handler" +
+                  f", chat_id: {update.message.chat_id}")
```
Comment (kmein) [inline bot.py]: This can be done on one line.

File: bot.py [hunk 3]
```diff
@@ -94,13 +104,15 @@ def send_menu(bot: Bot, chat_id: int, query: Query):
         )
 
 
-def menu_handler(bot: Bot, update: Update, args: List[str]):
-    text = demojize("".join(args))
+def menu_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: menu_handler" +
+                  f", chat_id: {update.message.chat_id}")
```
Comment (kmein) [inline bot.py]: Again, this can be done in one line.

File: bot.py [hunk 4]
```diff
@@ -120,7 +132,9 @@ def menu_handler(bot: Bot, update: Update, args: List[str]):
         )
 
 
-def info_handler(bot: Bot, update: Update):
+def info_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: info_handler" +
+                  f", chat_id: {update.message.chat_id}")
```
Comment (kmein) [inline bot.py]: 1 line pls

File: bot.py [hunk 5]
```diff
@@ -143,30 +157,36 @@ def info_handler(bot: Bot, update: Update):
     )
 
 
-def allergens_handler(bot: Bot, update: Update):
+def allergens_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: allergens_handler" +
+                  f", chat_id: {update.message.chat_id}")
```
Comment (kmein) [inline bot.py]: U know

File: bot.py [hunk 6]
```diff
@@ -143,30 +157,36 @@ def info_handler(bot: Bot, update: Update):
...
         update.message.chat_id,
         emojize("Wähle Deine Allergene aus. :index_pointing_up:"),
         reply_markup=allergens_chooser,
     )
 
 
-def resetallergens_handler(bot: Bot, update: Update):
+def resetallergens_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: resetallergens_handler" +
+                  f", chat_id: {update.message.chat_id}")
```
Comment (kmein) [inline bot.py]: ...

File: bot.py [hunk 7]
```diff
@@ -143,30 +157,36 @@ def info_handler(bot: Bot, update: Update):
...
+    context.bot.send_message(
         update.message.chat_id, emojize("Allergene zurückgesetzt. :heavy_check_mark:")
     )
 
 
-def mensa_handler(bot: Bot, update: Update, args: List[str]):
-    text = " ".join(args)
+def mensa_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: mensa_handler" +
+                  f", chat_id: {update.message.chat_id}")
```
Comment (kmein) [inline bot.py]: ...

File: bot.py [hunk 8]
```diff
@@ -230,58 +249,59 @@ def subscribe_handler(bot: Bot, update: Update, args: List[str], job_queue: JobQ
...
             else "Du hast dein Abonnement erfolgreich aktualisiert.",
         )
 
 
-def unsubscribe_handler(bot: Bot, update: Update, job_queue: JobQueue):
-    section = str(update.message.chat_id)
+def unsubscribe_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: status_handler" +
+                  f", chat_id: {update.message.chat_id}"
+                  f", is_subscriber: {user_db.is_subscriber(update.message.chat_id)}")
```
Comment (kmein) [inline bot.py]: One line, or maye something like
```python
", ".join(["Entering: status_handler", f"chat_id: {update.message.chat_id}", f"is_subscriber: {user_db.is_subscriber(update.message.chat_id)}"]
```

File: bot.py [hunk 9]
```diff
@@ -230,58 +249,59 @@ def subscribe_handler(bot: Bot, update: Update, args: List[str], job_queue: JobQ
...
 
-def status_handler(bot: Bot, update: Update, job_queue: JobQueue):
-    print(job_queue)
-    # section = str(update.message.chat_id)
-    bot.send_message(
+def status_handler(update: Update, context: CallbackContext):
+    logging.debug(f"Entering: status_handler" +
+                  f", chat_id: {update.message.chat_id}" +
+                  f", user_db.users(): {user_db.users()}" +
+                  f", user is_subscriber: {list(user for user in user_db.users() if user_db.is_subscriber(user))}")
```
Comment (kmein) [inline bot.py]: See above.

File: bot.py [hunk 10]
```diff
@@ -230,58 +249,59 @@ def subscribe_handler(bot: Bot, update: Update, args: List[str], job_queue: JobQ
...
+    context.bot.send_message(
         update.message.chat_id,
         f"Registered: {len(user_db.users())}\n"
-        f"Subscribed: {len(list(job for job in job_queue.jobs() if job.enabled))}",
+        f"Subscribed: {len(list(user for user in user_db.users() if user_db.is_subscriber(user)))}",
     )
 
 
+def notify_subscribers(context: CallbackContext):
+    logging.debug(f"Entering: notify_subscribers")
```
Comment (kmein) [inline bot.py]: No f-string necessary.

**Review by kmein — CHANGES_REQUESTED** [review]
Looks awesome, thanks for the good work.
Got only a few really tiny details for you to fix.

**Comment by qttylog** [comment]
I think I got everything :P
rdy for takeoff?

**Review by kmein — APPROVED** [review]

**Pull request merged**